{"uid":"d6843686400d3748","name":"test_build_and_deploy","fullName":"tests.integration.test_charm#test_build_and_deploy","historyId":"6e6e907ed99cb080c4e76760513358b7","time":{"start":1748925878207,"stop":1748925878207,"duration":0},"description":"Build the charm-under-test and deploy it together with related charms.\n\n    Assert on the unit status before any relations/configurations take place.\n    ","descriptionHtml":"<p>Build the charm-under-test and deploy it together with related charms.</p>\n<pre><code>Assert on the unit status before any relations/configurations take place.\n</code></pre>\n","status":"broken","statusMessage":"juju.errors.JujuConnectionError: Unable to connect to any endpoint: 10.152.183.30:17070","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 344, in from_call\n    result: TResult | None = func()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 246, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 512, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    raise exception\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 843, in pytest_runtest_setup\n    yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 53, in run_old_style_hookwrapper\n    return result.get_result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 103, in get_result\n    raise exc.with_traceback(tb)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/capture.py\", line 895, in pytest_runtest_setup\n    return (yield)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 164, in pytest_runtest_setup\n    item.session._setupstate.setup(item)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 514, in setup\n    col.setup()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1673, in setup\n    self._request._fillfixtures()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 719, in _fillfixtures\n    item.funcargs[argname] = self.getfixturevalue(argname)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 548, in getfixturevalue\n    fixturedef = self._get_active_fixturedef(argname)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 639, in _get_active_fixturedef\n    fixturedef.execute(request=subrequest)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1127, in execute\n    result = ihook.pytest_fixture_setup(fixturedef=self, request=request)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 512, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    raise exception\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 53, in run_old_style_hookwrapper\n    return result.get_result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 103, in get_result\n    raise exc.with_traceback(tb)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 53, in run_old_style_hookwrapper\n    return result.get_result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 103, in get_result\n    raise exc.with_traceback(tb)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1195, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 929, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 302, in _asyncgen_fixture_wrapper\n    result = event_loop.run_until_complete(setup())\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 284, in setup\n    res = await gen_obj.__anext__()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py\", line 265, in ops_test\n    await ops_test._setup_model()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py\", line 828, in _setup_model\n    await self._controller.connect(\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/controller.py\", line 104, in connect\n    await self._connector.connect_controller(controller_name, **kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connector.py\", line 113, in connect_controller\n    await self.connect(\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connector.py\", line 80, in connect\n    self._connection = await Connection.connect(**kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 350, in connect\n    raise lastError\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 339, in connect\n    await self._connect_with_redirect([_ep])\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 855, in _connect_with_redirect\n    login_result = await self._connect_with_login(endpoints)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 822, in _connect_with_login\n    await self._connect(endpoints)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 787, in _connect\n    raise errors.JujuConnectionError(\njuju.errors.JujuConnectionError: Unable to connect to any endpoint: 10.152.183.30:17070","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"event_loop","time":{"start":1748925878207,"stop":1748925878208,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"pytestconfig","time":{"start":1748925878208,"stop":1748925878208,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"tmp_path_factory","time":{"start":1748925878208,"stop":1748925878208,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1748925878208,"stop":1748925938319,"duration":60111},"status":"broken","statusMessage":"juju.errors.JujuConnectionError: Unable to connect to any endpoint: 10.152.183.30:17070\n","statusTrace":"  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 53, in run_old_style_hookwrapper\n    return result.get_result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 103, in get_result\n    raise exc.with_traceback(tb)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1195, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/_pytest/fixtures.py\", line 929, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 302, in _asyncgen_fixture_wrapper\n    result = event_loop.run_until_complete(setup())\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 284, in setup\n    res = await gen_obj.__anext__()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py\", line 265, in ops_test\n    await ops_test._setup_model()\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py\", line 828, in _setup_model\n    await self._controller.connect(\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/controller.py\", line 104, in connect\n    await self._connector.connect_controller(controller_name, **kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connector.py\", line 113, in connect_controller\n    await self.connect(\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connector.py\", line 80, in connect\n    self._connection = await Connection.connect(**kwargs)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 350, in connect\n    raise lastError\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 339, in connect\n    await self._connect_with_redirect([_ep])\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 855, in _connect_with_redirect\n    login_result = await self._connect_with_login(endpoints)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 822, in _connect_with_login\n    await self._connect(endpoints)\n  File \"/home/runner/work/operator-workflows/operator-workflows/tests/workflows/integration/test-upload-charm/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 787, in _connect\n    raise errors.JujuConnectionError(\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"afterStages":[{"name":"event_loop::2","time":{"start":1748925938344,"stop":1748925938344,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1748925938344,"stop":1748925938344,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1748925938344,"stop":1748925938344,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration"},{"name":"suite","value":"test_charm"},{"name":"host","value":"fv-az1121-948"},{"name":"thread","value":"24050-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.test_charm"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":true,"retry":true,"extra":{"categories":[],"tags":["asyncio"]},"source":"d6843686400d3748.json","parameterValues":[]}